CheckboxItem/CheckboxItem.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Agate styled item components with a toggleable checkbox.
+ *
+ * @example
+ * <CheckboxItem onToggle={console.log}>
+ * 	Item with a Checkbox
+ * </CheckboxItem>
+ *
+ * @module agate/CheckboxItem
+ * @exports CheckboxItem
+ * @exports CheckboxItemBase
+ */
+
+import kind from '@enact/core/kind';
+import PropTypes from 'prop-types';
+import React from 'react';
+import compose from 'ramda/src/compose';
+
+import Spottable from '@enact/spotlight/Spottable';
+import Toggleable from '@enact/ui/Toggleable';
+
+import Checkbox from '../Checkbox';
+import Item from '../Item';
+import Skinnable from '../Skinnable';
+
+import componentCss from './CheckboxItem.module.less';
+
+/**
+ * An item with a checkbox component, ready to use in Agate applications.
+ *
+ * `CheckboxItem` may be used to allow the user to select a single option or used as part of a
+ * [Group]{@link ui/Group} when multiple [selections]{@link ui/Group.Group.select} are possible.
+ *
+ * Usage:
+ * ```
+ * <CheckboxItem
+ * 	defaultSelected={selected}
+ * 	onToggle={handleToggle}
+ * >
+ *  Item with a Checkbox
+ * </CheckboxItem>
+ * ```
+ *
+ * @class CheckboxItem
+ * @memberof agate/CheckboxItem
+ * @extends agate/ToggleItem.ToggleItem
+ * @omit iconComponent
+ * @ui
+ * @public
+ */
+const CheckboxItemBase = kind({
+	name: 'CheckboxItem',
+
+	propTypes: /** @lends agate/CheckboxItem.CheckboxItem.prototype */ {
+		/**
+		 * Customizes the component by mapping the supplied collection of CSS class names to the
+		 * corresponding internal Elements and states of this component.
+		 *
+		 * The following classes are supported:
+		 *
+		 * * `checkboxItem` - The root class name
+		 *
+		 * @type {Object}
+		 * @public
+		 */
+		css: PropTypes.object,
+
+		/**
+		 * Customize the component used as the check in the checkbox.
+		 *
+		 * @type {Element|Function|String}
+		 * @default 'check'
+		 * @public
+		 */
+		icon: PropTypes.oneOfType([PropTypes.element, PropTypes.func, PropTypes.string]),
+
+		/**
+		 * Specifies on which side (`'before'` or `'after'`) of the text the checkbox appears.
+		 *
+		 * @type {('before'|'after')}
+		 * @default 'before'
+		 * @public
+		 */
+		iconPosition: PropTypes.string,
+
+		/**
+		 * If true the checkbox will be selected.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		selected: PropTypes.bool
+	},
+
+	defaultProps: {
+		icon: 'check',
+		iconPosition: 'before'
+	},
+
+	styles: {
+		css: componentCss,
+		className: 'checkboxItem',
+		publicClassNames: ['checkboxItem']
+	},
+
+	render: ({children, css, icon, iconPosition, selected, ...rest}) => (
+		<Item
+			data-webos-voice-intent="SelectCheckItem"
+			role="checkbox"
+			{...rest}
+			css={css}
+		>
+			{iconPosition === 'after' ? (<Checkbox selected={selected} slot="slotAfter">{icon}</Checkbox>) : null}
```

Comment:
We can simplify this and L116 by moving the conditional into `slot` since the order isn't relevant when using `Slottable`; it'll pluck them into the right slot regardless of order.

```jsx
<Checkbox selected={selected} slot={iconPosition === 'after' ? 'slotAfter' : 'slotBefore'}>{icon}</Checkbox>
```